src/forms/Button.jsx

```diff
@@ -1,105 +1,42 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import cx from 'classnames';
-import Flex from '../layout/Flex';
-import FlexItem from '../layout/FlexItem';
 
-export const BUTTON_CLASS = 'button';
-export const BUTTON_ICON_WRAPPER_CLASS = 'button-icon-wrapper';
-export const BUTTON_LABEL_CLASS = 'button-label';
-export const BUTTON_ICON_CLASS = 'button-icon';
+import { Button as SwarmButton } from '@meetup/swarm-components';
 
 /**
  * @module Button
  */
 class Button extends React.PureComponent {
 	render() {
 		const {
-			children,
-			className,
-			reset,
-			fullWidth,
-			primary,
-			neutral,
+			// removing reset & fullWidth props from other to prevent an invalid boolean being passed to <button>
+			reset, // eslint-disable-line no-unused-vars
+			fullWidth, // eslint-disable-line no-unused-vars
 			icon,
-			right,
-			small,
-			bordered,
-			hasHoverShadow,
+			hasHoverShadow, // eslint-disable-line no-unused-vars
 			component,
-			disabled,
-			type,
 			...other
 		} = this.props;
-		const classNames = {
-			button: cx(
-				BUTTON_CLASS,
-				{
-					'button--fullWidth': fullWidth,
-					'button--primary': primary,
-					'button--small': small,
-					'button--reset': reset,
-					'button--bordered': bordered,
-					'button--hasHoverShadow': hasHoverShadow,
-					'button--neutral': neutral,
-					'button--disabled': disabled,
-					'button--iconOnly': icon && !children,
-				},
-				className
-			),
-			iconWrap: cx(BUTTON_ICON_WRAPPER_CLASS, {
-				[`${BUTTON_ICON_WRAPPER_CLASS}--right`]: right,
-			}),
-		};
-		const opts = right ? { rowReverse: 'all' } : {};
 
-		const iconChildren = (
-			<Flex className={classNames.iconWrap} justify="center" {...opts}>
-				{icon && (
-					<FlexItem
-						shrink
-						className={`${BUTTON_ICON_CLASS} valign--middle flush--left`}
-					>
-						{icon}
-					</FlexItem>
-				)}
-				{children && (
-					<FlexItem
-						shrink
-						className={`${BUTTON_LABEL_CLASS} valign--middle align--center atMedium_align--left`}
-					>
-						{children}
-					</FlexItem>
-				)}
-			</Flex>
-		);
-
-		const Component = component;
+		// checking for the icon component signature if passed as icon={<Icon shape="search"/>} and grabbing the shape prop
+		// since this functionality is built into button in order to enforce strict sizes on button icons
+		const props = { icon: icon && icon.props && icon.props.shape };
 
-		if (disabled) {
-			delete other.onClick;
+		if (component !== 'button') {
+			console.warn(
+				'All components are button elements. Future iterations will support links as independent components and the logic will be handled here'
```

Comment:
```suggestion
				'All Swarm UI v2 components are button elements. Future iterations will support links as independent components and the logic will be handled here'
```

Reply:
To continue off of Jackie's comment -- how do we want to handle `LinkButton`? There's a PR in swarm UI. I assume we want to continue supporting anchor tag buttons, so this would would need to be updated to return `LinkButton` or `Button`.

Reply:
We should circle back to this and update it use the `LinkButton`